scripts/globals/additional_effects.lua

```diff
@@ -0,0 +1,263 @@
+require("scripts/globals/magic") -- For resist functions
+require("scripts/globals/utils") -- For clamping function
+require("scripts/globals/msg")
+--------------------------------------
+
+tpz = tpz or {}
+tpz.addEffect = tpz.addEffect or {}
+
+tpz.addEffect.isRanged = function(item)
+    -- Archery/Marksmanship/Throwing
+    return math.abs(item:getSkillType() - tpz.skill.MARKSMANSHIP) < 2
+end
+
+tpz.addEffect.calcRangeBonus = function(attacker, defender, element, damage)
+    -- Copied from existing scripts.
+    local bonus = 0
+    if element == tpz.magic.ele.LIGHT then
+        bonus = attacker:getStat(tpz.mod.MND) - defender:getStat(tpz.mod.MND)
+        if bonus > 40 then
+            bonus = bonus + (bonus -40) /2;
+            damage = damage + bonus
+        end
+    else
+        bonus = attacker:getStat(tpz.mod.INT) - defender:getStat(tpz.mod.INT)
+        if bonus > 20 then
+            bonus = bonus + (bonus -20) /2;
+            damage = damage + bonus
+        end
+    end
+
+    return damage
+end
+
+tpz.addEffect.levelCorrection = function(dLV, aLV, chance)
+    -- Level correction of proc chance (copied from existing bolt/arrow scripts, looks wrong..)
+    if dLV > aLV then
+        chance = utils.clamp(chance - 5 * (dLV - aLV), 5, 95)
+    end
+
+    return chance
+end
+
+tpz.addEffect.statusAttack = function(addStatus)
+    local effectList =
+    {
+        [tpz.effect.DEFENSE_DOWN] = {tick = 0, strip = tpz.effect.DEFENSE_BOOST},
+        [tpz.effect.EVASION_DOWN] = {tick = 0, strip = tpz.effect.EVASION_BOOST},
+        [tpz.effect.ATTACK_DOWN]  = {tick = 0, strip = tpz.effect.ATTACK_BOOST},
+        [tpz.effect.POISON]       = {tick = 3, strip = nil},
+        [tpz.effect.CHOKE]        = {tick = 3, strip = nil},
+    }
+    local delEffect = effectList[addStatus]
+    defender:delStatusEffect(delEffect.strip)
+    return tick
+end
+
+tpz.addEffect.calcDamage = function(attacker, element, defender, damage)
+    local params = {}
+    params.bonusmab = 0
+    params.includemab = false
+    damage = addBonusesAbility(attacker, element, defender, damage, params)
+    damage = damage * applyResistanceAddEffect(attacker, defender, element, 0)
+    damage = adjustForTarget(defender, damage, element)
+    damage = finalMagicNonSpellAdjustments(attacker, defender, element, damage)
+    return damage
+end
+
+-- paralyze on hit, fire damage on hit, etc..
+function additionalEffectAttack(attacker, defender, baseAttackDamage, item)
+    local addType = item:getMod(tpz.mod.ITEM_ADDEFFECT_TYPE)
+    local subEffect = item:getMod(tpz.mod.ITEM_SUBEFFECT)
+    local damage = item:getMod(tpz.mod.ITEM_ADDEFFECT_DMG)
+    local chance = item:getMod(tpz.mod.ITEM_ADDEFFECT_CHANCE)
+    local element = item:getMod(tpz.mod.ITEM_ADDEFFECT_ELEMENT)
+    local addStatus = item:getMod(tpz.mod.ITEM_ADDEFFECT_STATUS)
+    local power = item:getMod(tpz.mod.ITEM_ADDEFFECT_POWER)
+    local duration = item:getMod(tpz.mod.ITEM_ADDEFFECT_DURATION)
```

Comment:
Is duration of a status effect always the same when an AE procs? There's no random element?

Reply:
some may get modified by resistance. this is unknown. The current scripts in release are using static duration's with zero modification to them.